pkg/snapshot/publisher.go

```diff
@@ -18,21 +18,29 @@ package snapshot
 import (
 	"bytes"
 	"fmt"
+	"sync/atomic"
+	"time"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/jmoiron/sqlx"
 	"github.com/multiformats/go-multihash"
+	"github.com/sirupsen/logrus"
 
 	"github.com/ethereum/go-ethereum/statediff/indexer/ipfs/ipld"
 	"github.com/ethereum/go-ethereum/statediff/indexer/postgres"
 	"github.com/ethereum/go-ethereum/statediff/indexer/shared"
 )
 
+const logInterval = 1 * time.Minute
+
 // Publisher is wrapper around DB.
 type Publisher struct {
 	db            *postgres.DB
 	currBatchSize uint
+	stateNodeCounter   uint64
+	storageNodeCounter   uint64
+	blockNodeCounter   uint64
```

Comment:
`blockNodeCounter` == `stateNodeCounter` + `storageNodeCounter` + the number of codeNodes.

So rather than having a `blockNodeCounter` let's have `codeNodeCounter`.